### PR TITLE
[SYCL] Add support for L0 loader validation layer

### PR DESCRIPTION
**Pull request by kholland-intel**
Extended debugging capabilities by setting two new environment variables (
ZE_ENABLE_VALIDATION_LAYER and ZE_ENABLE_PARAMETER_VALIDATION) that will enable
the Level Zero loader validation layer.

Also, implemented ZE_DEBUG as a bit mask.

### CODE REVIEW

File: sycl/plugins/level_zero/pi_level_zero.cpp [hunk 1]
```diff
@@ -589,6 +610,11 @@ pi_result piPlatformsGet(pi_uint32 NumEntries, pi_platform *Platforms,
   static std::vector<pi_platform> PiPlatformsCache;
   static std::mutex PiPlatformsCacheMutex;
 
+  if (ZeValidationLayer) {
```
Comment (smaslov-intel) [inline sycl/plugins/level_zero/pi_level_zero.cpp:613]: Please add a comment that having these set in the environment **before** running zeInit will enable validation layer in Level-Zero loader.

File: sycl/plugins/level_zero/pi_level_zero.cpp [hunk 2]
```diff
@@ -25,6 +25,10 @@
 
 #include "usm_allocator.hpp"
 
+#ifdef _WIN32
+#define setenv(name, value, overwrite) _putenv_s(name, value)
```
Comment (smaslov-intel) [inline sycl/plugins/level_zero/pi_level_zero.cpp:29]: Let's not use macros without a good reason. Please just create a new function that works for both Linux and Windows.

**Review by smaslov-intel — APPROVED** [review]
Approved to get testing started

**Comment by bader** [comment]
@kholland-intel, could you resolve merge conflict, please?

File: sycl/plugins/level_zero/pi_level_zero.cpp [hunk 3]
```diff
@@ -580,6 +578,16 @@ static bool isOnlineLinkEnabled();
 // End forward declarations for mock Level Zero APIs
 std::once_flag OnceFlag;
 
+// This function will ensure compatibility with both Linux and Windowns for
+// setting environment variables.
+static void setEnvVar(const char *name, const char *value) {
```
Comment (smaslov-intel) [inline sycl/plugins/level_zero/pi_level_zero.cpp:583]: I suggest checking error-code and return bool here, indicating if var was set, even if currently you aren't going to use it.

**Review by smaslov-intel — DISMISSED** [review]
minor comment about return value from setEnvVar

**Review by smaslov-intel — APPROVED** [review]

**Pull request merged**